### PR TITLE
PWN-4266 - add GoogleSignInHelper.kt and WalletAuthManager.kt to make Auth

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-4266

## Description of Work

Add GoogleSignInHelper.kt and WalletAuthManager.kt to make Auth

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/auth/common/GoogleSignInHelper.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        getSignInClient(context).apply {
+            signOut()
+            getSignInIntent(request)
+                .addOnSuccessListener {
+                    googleSignInLauncher.launch(IntentSenderRequest.Builder(it).build())
+                }
+        }
+    }
+
+    fun handleSignResult(context: Context, result: ActivityResult): SignInCredential? {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/common/GoogleSignInHelper.kt:37]: how's `handleSignInResult`
or parseSignInResult, handle doesn't really give you an idea when SignInCredential is returned null.

File: app/src/main/java/org/p2p/wallet/auth/common/WalletAuthManager.kt [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+import android.webkit.JavascriptInterface
+import android.webkit.WebView
+import android.widget.Toast
+import com.google.gson.Gson
+import org.p2p.wallet.auth.model.DeviceShareKey
+import org.p2p.wallet.auth.model.GoogleFlow
+import org.p2p.wallet.infrastructure.security.SecureStorageContract
+
+private const val KEY_DEVICE_SHARE = "KEY_DEVICE_SHARE"
+private const val indexUrl = "file:///android_asset/index.html"
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/common/WalletAuthManager.kt:15]: indexUri

File: app/src/main/res/values/environment_config.xml [hunk 3]
```diff
@@ -27,4 +27,6 @@
 
     <string name="notificationServiceBaseUrl">https://push-service.wallet.p2p.org/</string>
     <string name="notificationServiceTestBaseUrl">http://35.234.120.240:9090/</string>
+
+    <string name="googleClientId">553127941597-j88bgo2invnpiefq2j1fhq8duju6o2tq.apps.googleusercontent.com</string>
```
Comment (gslevinkov) [inline app/src/main/res/values/environment_config.xml:31]: looks like secret info for me, no?

File: app/src/main/java/org/p2p/wallet/auth/common/WalletAuthManager.kt [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    fun onSignIn(idToken: String, deviceShare: String? = null) {
+        deviceShare?.let {
+            onboardingWebView.evaluateJavascript(
+                "new p2pWeb3Auth.AndroidFacade().triggerSignInNoCustom('$idToken', $it)",
+                null
+            )
+        } ?: onboardingWebView.evaluateJavascript(
+            "new p2pWeb3Auth.AndroidFacade().triggerSignInNoDevice('$idToken')",
+            null
+        )
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/common/WalletAuthManager.kt:85]:
```suggestion
        deviceShare
          ?.let { "new p2pWeb3Auth.AndroidFacade().triggerSignInNoCustom('$idToken', $it)" } 
          ?: "new p2pWeb3Auth.AndroidFacade().triggerSignInNoDevice('$idToken')"
          ?.also { onboardingWebView.evaluateJavascript(it, null) } 
```

File: app/src/main/java/org/p2p/wallet/auth/common/WalletAuthManager.kt [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        }
+    }
+
+    fun setIdToken(userId: String, idToken: String) {
+        lastUserId = userId
+        lastIdToken = idToken
+        if (flowMode == GoogleFlow.SIGN_UP) {
+            onSignUp(idToken)
+        } else {
+            if (hasDeviceShare()) {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/common/WalletAuthManager.kt:55]: isDeviceSharedSaved?

File: app/src/main/res/values/environment_config.xml (hunk 3 above)
Comment (kabramovp2p) [reply app/src/main/res/values/environment_config.xml:31]: yes we should pass it somewhere in build config for example
what can you suggest?

File: app/src/main/java/org/p2p/wallet/auth/common/WalletAuthManager.kt [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+        onboardingWebView.apply {
+            loadUrl(indexUrl)
+            addJavascriptInterface(
+                AndroidCommunicationChannel(context),
+                androidCommunicationChannel
+            )
+        }
+    }
+
+    fun setIdToken(userId: String, idToken: String) {
```
[the remark this answers is not in the document]
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/auth/common/WalletAuthManager.kt:49]: it is idToken auth token

File: app/src/main/res/values/environment_config.xml (hunk 3 above)
Comment (gslevinkov) [reply app/src/main/res/values/environment_config.xml:31]: extract it to CI

File: app/src/main/java/org/p2p/wallet/auth/common/WalletAuthManager.kt (hunk 6 above)
[the remark this answers is not in the document]
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/auth/common/WalletAuthManager.kt:49]: authIdToken then?
idToken is too abstract to use without any context

File: app/src/main/java/org/p2p/wallet/auth/common/WalletAuthManager.kt [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+import com.google.gson.Gson
+import org.p2p.wallet.auth.model.DeviceShareKey
+import org.p2p.wallet.auth.model.GoogleAuthFlow
+import org.p2p.wallet.infrastructure.security.SecureStorageContract
+
+private const val KEY_DEVICE_SHARE = "KEY_DEVICE_SHARE"
+private const val indexUri = "file:///android_asset/index.html"
+private const val communicationChannel = "AndroidCommunicationChannel"
+
+class WalletAuthManager(
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/common/WalletAuthManager.kt:18]: you can put some impl details. how this auth works
smth like
WalletWeb3AuthManager

**Review by gslevinkov — APPROVED** [review]

**Review by kuznetsov — APPROVED** [review]
🔥

**Pull request merged**